lib/client.js

```diff
@@ -347,6 +352,20 @@ export default class Client {
       this.ready = true
       this._metadata = data.metadata
       this._context = data.context
+      this._iframe_session_timeout = data.iframe_session_timeout
+      this._idleState = new IdleState(data.iframe_session_timeout)
+      // Add an observer.  This will be called when the idleState changes
+      this._idleState.addObserver(state => {
+        if (state === 'active') {
+          // ... application is now active, timeout clock is reset to zero
+          // Send a postMessage indicating liveness + possibly a payload too
+          this.postMessage('session.live')
+        } else if (state === 'idle') {
+          // ... application is now idle, timeout clock is ticking
+          // Send a postMessage indicating sessionTimeout + possibly a payload too
+          this.postMessage('session.idle')
+        }
+      })
```

Comment:
putting this in a try catch block, just reduces risk to LOW i think. What do you think @token-cjg ?

Reply:
That is possible, if we create a new change and rewrite this in said change so that we're not switching up a number of the core packages such as Babel + introducing TypeScript, then try catch this, this change could become low risk.

Just introducing try catch here though without creating a fresh PR to reduce the amount of shiny things however will not decrease the risk too much.